controllers/storagecluster/cephfilesystem.go

```diff
@@ -19,59 +20,73 @@ type ocsCephFilesystems struct{}
 
 // newCephFilesystemInstances returns the cephFilesystem instances that should be created
 // on first run.
-func (r *StorageClusterReconciler) newCephFilesystemInstances(initData *ocsv1.StorageCluster) ([]*cephv1.CephFilesystem, error) {
+func (r *StorageClusterReconciler) newCephFilesystemInstances(initStorageCluster *ocsv1.StorageCluster) ([]*cephv1.CephFilesystem, error) {
 	ret := &cephv1.CephFilesystem{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      generateNameForCephFilesystem(initData),
-			Namespace: initData.Namespace,
+			Name:      generateNameForCephFilesystem(initStorageCluster),
+			Namespace: initStorageCluster.Namespace,
 		},
 		Spec: cephv1.FilesystemSpec{
 			MetadataPool: cephv1.PoolSpec{
-				Replicated:    generateCephReplicatedSpec(initData, "metadata"),
-				FailureDomain: initData.Status.FailureDomain,
+				Replicated:    generateCephReplicatedSpec(initStorageCluster, "metadata"),
+				FailureDomain: initStorageCluster.Status.FailureDomain,
 			},
 			MetadataServer: cephv1.MetadataServerSpec{
 				ActiveCount:   1,
 				ActiveStandby: true,
-				Placement:     getPlacement(initData, "mds"),
-				Resources:     defaults.GetDaemonResources("mds", initData.Spec.Resources),
+				Placement:     getPlacement(initStorageCluster, "mds"),
+				Resources:     defaults.GetDaemonResources("mds", initStorageCluster.Spec.Resources),
 				// set PriorityClassName for the MDS pods
 				PriorityClassName: openshiftUserCritical,
 			},
 		},
 	}
 
-	if initData.Spec.StorageProfiles == nil {
-		// standalone deployment will not have storageProfile, we need to
-		// define default dataPool, if storageProfile is set this will be
-		// overridden.
-		ret.Spec.DataPools = []cephv1.NamedPoolSpec{
-			{
-				PoolSpec: cephv1.PoolSpec{
-					DeviceClass:   generateDeviceClass(initData),
-					Replicated:    generateCephReplicatedSpec(initData, "data"),
-					FailureDomain: initData.Status.FailureDomain,
+	// Load all StorageProfile objects in the StorageCluster's namespace
+	storageProfiles := &ocsv1.StorageProfileList{}
+	spErr := r.Client.List(r.ctx, storageProfiles, &client.ListOptions{Namespace: initStorageCluster.GetNamespace()})
```

Comment:
- I don't see a reason for different name
```suggestion
	err := r.Client.List(r.ctx, storageProfiles, &client.ListOptions{Namespace: initStorageCluster.GetNamespace()})
```

Reply:
@leelavg there's already a variable with the name `err` in this [block](https://github.com/raaizik/ocs-operator/blob/dba040ca8f756a6d8e49c7c1596d75272feb9811/controllers/storagecluster/cephfilesystem.go#L89).

Reply:
ack, anyways we are also touching that line, maybe have `err` at both places.